processor/probabilisticsamplerprocessor/sampler_mode.go

```diff
@@ -312,6 +504,24 @@ func commonShouldSampleLogic[T any](
 
 	sampled := threshold.ShouldSample(rnd.randomness())
 
+	if sampled && carrier != nil {
+		// Note: updateThreshold limits loss of adjusted count, by
+		// preventing the threshold from being lowered, only allowing
+		// probability to fall and never to rise.
+		if err := carrier.updateThreshold(threshold); err != nil {
+			if errors.Is(err, sampling.ErrInconsistentSampling) {
+				// This is working-as-intended.  You can't lower
+				// the threshold, it's illogical.
+				logger.Debug(description, zap.Error(err))
+			} else {
+				logger.Warn(description, zap.Error(err))
+			}
+		}
+		if err := carrier.reserialize(); err != nil {
+			logger.Info(description, zap.Error(err))
```

Comment:
and same here :-)

Reply:
This is some sort of grievous corruption, and I personally do not want every component inventing new metrics to monitor for things we never expect to happen.